godot-core/src/engine/gfile.rs

```diff
@@ -662,33 +662,25 @@ impl Read for GFile {
             return Ok(0);
         }
 
-        let mut read_bytes = 0;
-        while read_bytes < bytes_to_read {
-            buf[read_bytes] = self.fa.get_8();
-            read_bytes += 1;
-            self.check_error()?;
-        }
+        let gd_buffer = self.fa.get_buffer(bytes_to_read as i64);
+        buf[0..gd_buffer.len()].copy_from_slice(gd_buffer.as_slice());
+
+        self.check_error()?;
 
-        Ok(read_bytes)
+        Ok(gd_buffer.len())
     }
 }
 
 impl Write for GFile {
     fn write(&mut self, buf: &[u8]) -> IoResult<usize> {
-        // TODO: Refactor to make use of `FileAccess::store_buffer()` method for higher efficiency in writing to file.
-        let bytes_to_write = buf.len();
-        let mut bytes_written = 0;
-
-        while bytes_written < bytes_to_write {
-            self.fa.store_8(buf[bytes_written]);
-            bytes_written += 1;
-            self.check_error()?;
-        }
-        Ok(bytes_written)
+        self.fa.store_buffer(PackedByteArray::from(buf));
+        self.check_error()?;
+
+        Ok(buf.len())
     }
 
+    // No-op, as still there is no need in `write` to keep anything in its internal buffer.
     fn flush(&mut self) -> IoResult<()> {
-        // TODO: After refactoring `write` implementation, assess if this method can stay as no-opt.
         Ok(())
     }
 }
```

Comment:
I wonder if we should not call [`FileAccess::flush()`](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-flush) here? 

I'm not sure if `store_buffer` flushes after each call, the docs don't say such a thing. Explicit flush would help in cases where the `GFile` is not properly dropped (e.g. in case of non-graceful program exit).

Reply:
Don't see any reason why we shouldn't call it there, will add it